brigade.js

```diff
@@ -207,13 +216,24 @@ events.on("push", (e, p) => {
     // This is an official release with a semantically versioned tag
     let matchTokens = Array.from(matchStr);
     let version = matchTokens[1];
-    return Group.runAll([
-      goTest(),
-      jsTest(),
-      yarnAudit(),
-      e2e()
-    ])
-    .then(() => {
+    // Important: To prevent Promise.all() from failing fast, we catch and
+    // return all errors. This ensures Promise.all() always resolves. We then
+    // iterate over all resolved values looking for errors. If we find one, we
+    // throw it so the whole build will fail.
+    //
+    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#Promise.all_fail-fast_behaviour
+    Promise.all([
+      goTest().run().catch((err) => { return err }),
+      jsTest().run().catch((err) => { return err }),
+      // Swallow audit errors because we want to treat these as informational only
+      // and do not want them to interfere with progressing further.
+      yarnAudit().run().catch(() => { return null }),
+      e2e().run().catch((err) => { return err })
+    ]).then((values) => {
+      values.forEach((value) => {
+        if (value instanceof Error) throw value;
+      });
+    }).then(() => {
```

Comment:
Applying this pattern permits jobs running concurrently with `yarnAudit` to run to completion, even though the `yarnAudit` job will continue to fail in the near term.